backend/dataline/services/connection.py

```diff
@@ -270,3 +270,55 @@ async def create_sas7bdat_connection(self, session: AsyncSession, file: UploadFi
         finally:
             # Clean up the temporary file
             os.unlink(temp_file_path)
+
+    async def refresh_connection_schema(self, session: AsyncSession, connection_id: UUID) -> ConnectionOut:
+        """
+        Refresh the schema of a connection. Flow of the function:
+        1. Get the connection details from the database
+        2. Get the latest schema information from the database (using DatalineSQLDatabase)
+        3. Create new ConnectionOptions with schema information from step 2
+        4. Fetch stored ConnectionOptions from the database and, if it exists, merge with new schema information
+        5. Sort schemas and tables by name
+        6. Update the connection with new options
+        """
+        connection = await self.connection_repo.get_by_uuid(session, connection_id)
+
+        # Get the latest schema information
+        db = await self.get_connection_details(connection.dsn)
```

Comment:
should we rename this to get_db_from_connection or something?

connection details seems a bit too vague to me now